### PR TITLE
Change FinalizerThreadCreate location to after profiler is initialized

### PR DESCRIPTION
**Pull request by sywhang**
Changed FinalizeThreadCreate to be called after ProfilerAPI is initialized to ensure finalizer creation notification to profilerAPI.

Fix issue #13499

### CODE REVIEW

File: src/vm/ceemain.cpp [hunk 1]
```diff
@@ -956,6 +952,10 @@ void EEStartupHelper(COINITIEE fFlags)
         // throws on error
         SetupThread();
 
+        // This isn't done as part of InitializeGarbageCollector() above because thread
+        // creation requires AppDomains to have been set up.
+        FinalizerThread::FinalizerThreadCreate();
+
```
Comment (noahfalk) [inline src/vm/ceemain.cpp]: How about we move this down to line 1011? On desktop there is a function InitializeGarbageCollection that contains the code:

    FinalizerThread::FinalizerThreadCreate();
    SetGarbageCollectorFullyInitialized();

It appears that function was refactored as part of the local GC work, but I would keep the ordering as similar as possible unless you've already tested that option and found it caused a problem.

Comment (noahfalk) [reply src/vm/ceemain.cpp]: @Maoni0 @swgillespie - Adding the GC folks in case they want to weigh in any further.

Comment (Maoni0) [reply src/vm/ceemain.cpp]: seems better to me.

Comment (swgillespie) [reply src/vm/ceemain.cpp]: looks fine to me.

**Pull request merged**